Automattic/wp-calypso
# Gutenbording launch flow improvements

#### Changes proposed in this Pull Request
* Clear Free plan selection when a custom domain is selected.
* When first pressing Complete Setup button inside the editor, start the flow at the first incomplete step.
* Clear the domain picker initial search query if it's the default site title ("Site Title" for EN).
* Disable _Continue_ button on _Select a domain_ step when search query is empty.
* Use site title as fallback to always have a domain suggestion on the summary step.
* Keep the domain step completed after an action has been made (selecting subdomain or pressing Confirm button).
* Update step completion to be derived from state instead on saved in Launch store. Should be no more errors.

#### Testing instructions
* Create a site using /new
* Apply this patch and sandbox your site 
* Go to editor, start the launch flow using Complete Setup button and check all of the above

Fixes part of #43750
